__tests__/Unit/Components/Tasks/TaskDetails.test.tsx

```diff
@@ -659,3 +643,78 @@ describe('TaskDependency', () => {
         });
     });
 });
+
+jest.mock('next/router', () => ({
```

Comment:
why this is added in middle of tests?

Reply:
The mock for `next/router` was added in the middle of the tests to ensure that the component being tested behaves consistently regardless of the actual behaviour of the router. By mocking the router, we can control its behaviour and provide specific values or functions that our component relies on, making the tests more predictable and isolated from external dependencies. This approach helps maintain the stability and reliability of the tests, ensuring that they remain consistent even if changes are made to the router implementation.